src/interfaces/FunnelErrors.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+/// @title Errors for Funnel contracts and FunnelFactory
+/// @author Zac (zlace0x), zhongfu (zhongfu), Edison (edison0xyz)
+interface FunnelErrors {
```

Comment:
should it be IFunnelErrors? Or we can do without an interface too

See:
https://github.com/Uniswap/permit2/blob/main/src/PermitErrors.sol

Reply:
I'm not sure but seaport seems to do it under Interfaces without naming it as `I`. But for consistency we can just go with one or another. https://github.com/ProjectOpenSea/seaport/blob/main/contracts/interfaces/TransferHelperErrors.sol